tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_BjorhusImpl.cpp

```diff
@@ -0,0 +1,157 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Side.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+using frame = Frame::Inertial;
+constexpr size_t VolumeDim = 3;
+
+// Test boundary conditions on dt<VSpacetimeMetric>
+void test_constraint_preserving_bjorhus_v_psi(
+    const size_t grid_size_each_dimension) noexcept {
+  // Setup grid
+  Mesh<VolumeDim> mesh{grid_size_each_dimension, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  const Direction<VolumeDim> direction(1, Side::Upper);  // +y direction
+  const size_t slice_grid_points =
+      mesh.extents().slice_away(direction.dimension()).product();
+
+  // Populate various tensors needed to compute BcDtVSpacetimeMetric
+  tnsr::iaa<DataVector, VolumeDim, frame> local_three_index_constraint(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  tnsr::I<DataVector, VolumeDim, frame> local_unit_interface_normal_vector(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  tnsr::aa<DataVector, VolumeDim, frame> local_char_projected_rhs_dt_v_psi(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  std::array<DataVector, 4> local_char_speeds{
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN()),
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN()),
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN()),
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN())};
+  {
+    // Setting 3idxConstraint
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      for (size_t a = 0; a <= VolumeDim; ++a) {
+        for (size_t b = 0; b <= VolumeDim; ++b) {
+          local_three_index_constraint.get(0, a, b)[i] = 11. - 3.;
```

Comment:
Why not just `8.` on the RHS?  Similarly below.

Reply:
In SpEC, this constraint is explicitly computed using d_i psi_ab and phi_iab as inputs. The explicit subtractions below are here to remind us what values those two input tensors were set to in SpEC to get the desired BC in this test. I've added this explanation as a code comment.